ios/RNZendesk/RNZendesk.swift

```diff
@@ -214,13 +214,20 @@ class RNZendesk: RCTEventEmitter {
                 }
 
                 if result != nil {
-                    var requestDicts = result!.requests.map { (request: ZDKRequest) -> [String: String] in
-                        var requestDict : [String: String] = [
+                    let requestDicts = result!.requests.map { (request: ZDKRequest) -> [String: Any] in
+                        let requestDict : [String: Any] = [
                             "id" : request.requestId,
                             "status" : request.status,
-                            "subject" : request.subject!,
+                            "subject" : request.subject ?? "",
                             "updatedAt": "\(Int(request.updateAt.timeIntervalSince1970) * 1000)",
-                            "lastComment": request.lastComment!.body
+                            "lastComment": request.lastComment!.body ?? "",
+                            "avatarUrls": request.commentingAgentsIds.map { (id: NSNumber)  in
+                                let agent = result!.commentingAgents.first { agent in
```

Comment:
can we not just map over `result.commentingAgents` directly?

feels a bit weird to map over a list of ids, and then find the agent with that id in an array

Reply:
> can we not just map over `result.commentingAgents` directly?
> 
> feels a bit weird to map over a list of ids, and then find the agent with that id in an array

`result` isn't the `request`. `result.commentingAgents` is the list of agents for _every_ Zendesk request, not a specific one.

Reply:
Yes, it is a bit weird but that's the interface we're provided by zendesk.

In the iOS SDK we only have the commenting agents list for _all_ of the requests that were returned, not on a request-by-request basis. So here we are consolidating the list of IDs that are on a request-by-request basis with the whole list.

Reply:
> `result` isn't the `request`. `result.commentingAgents` is the list of agents for _every_ Zendesk request, not a specific one.

gotcha!